### PR TITLE
Scannetppv2 Configs

### PR DESCRIPTION
**Pull request by Streakfull**
### Main Changes

* Added 3 new config files for the scannetpp dataset.
   * CAC
   * OA-CNN
   * OctFormer
* Minor changes in existing scannetpp config files.
---
### Bug Fixes
* Resolved an issue where training would crash if the input sample segment contained only ignored labels (cross-entropy loss error) → now this step is bypassed.
* The `verify=True` parameter for `FormatCode` has been removed. See changelog [here](https://github.com/google/yapf/blob/main/CHANGELOG.md#0402-2023-09-22).
---
*Note: There are some detailed comments on specific lines.*

### CODE REVIEW

File: configs/scannetpp/semseg-spunet-v1m1-0-base.py [hunk 1]
```diff
@@ -20,7 +20,10 @@
         channels=(32, 64, 128, 256, 256, 128, 96, 96),
         layers=(2, 3, 4, 6, 2, 2, 2, 2),
     ),
-    criteria=[dict(type="CrossEntropyLoss", loss_weight=1.0, ignore_index=-1)],
+    criteria=[
+        dict(type="CrossEntropyLoss", loss_weight=1.0, ignore_index=-1),
+        dict(type="LovaszLoss", mode="multiclass", loss_weight=1.0, ignore_index=-1),
```
Comment (Streakfull) [inline configs/scannetpp/semseg-spunet-v1m1-0-base.py:25]: LovaszLoss added in the criteria here to be consistent with our other experiments. Let me know if this config should be in a seperate file. i.e `configs/scannetpp/semseg-spunet-v1m1-0-lovasz.py`.

File: pointcept/engines/hooks/evaluator.py [hunk 2]
```diff
@@ -150,18 +152,23 @@ def eval(self):
             self.trainer.storage.put_scalar("val_intersection", intersection)
             self.trainer.storage.put_scalar("val_union", union)
             self.trainer.storage.put_scalar("val_target", target)
-            self.trainer.storage.put_scalar("val_loss", loss.item())
+            self.trainer.storage.put_scalar(
+                "val_loss", loss.item() if loss is not None else 0
```
Comment (Streakfull) [inline pointcept/engines/hooks/evaluator.py:156]: Handles the case when the step is skipped and the loss is None.

File: pointcept/engines/hooks/evaluator.py [hunk 3]
```diff
@@ -176,7 +183,12 @@ def eval(self):
                 m_iou, m_acc, all_acc
             )
         )
+        current_epoch = self.trainer.epoch + 1
         for i in range(self.trainer.cfg.data.num_classes):
+            name = self.trainer.cfg.data.names[i]
+            self.trainer.writer.add_scalar(
+                f"val/cls_{i}-{name} Iou", iou_class[i], current_epoch
```
Comment (Streakfull) [inline pointcept/engines/hooks/evaluator.py:190]: Additional logging class/iou.

File: pointcept/engines/train.py [hunk 4]
```diff
@@ -263,7 +269,7 @@ def build_train_loader(self):
             collate_fn=partial(point_collate_fn, mix_prob=self.cfg.mix_prob),
             pin_memory=True,
             worker_init_fn=init_fn,
-            drop_last=True,
+            drop_last=len(train_data) > self.cfg.batch_size_val_per_gpu,
```
Comment (Streakfull) [inline pointcept/engines/train.py:272]: This was causing issues during our overfitting experiments.

File: pointcept/utils/config.py [hunk 5]
```diff
@@ -493,7 +493,7 @@ def _format_dict(input_dict, outest_level=False):
             blank_line_before_nested_class_or_def=True,
             split_before_expression_after_opening_paren=True,
         )
-        text, _ = FormatCode(text, style_config=yapf_style, verify=True)
```
Comment (Streakfull) [inline pointcept/utils/config.py:496]: https://github.com/google/yapf/blob/main/CHANGELOG.md#0402-2023-09-22

File: pointcept/engines/hooks/evaluator.py (hunk 2 above)
Comment (Gofinge) [reply pointcept/engines/hooks/evaluator.py:156]: Hey @Streakfull,

I think if data doesn't contain a valid label, maybe we should filter is out during the data preprocessing. Maybe using a JSON file to record valid data is also a good way. Skipping the training step might also cause some problems like the scheduler is not updated.

File: pointcept/engines/train.py (hunk 4 above)
Comment (Gofinge) [reply pointcept/engines/train.py:272]: @Streakfull 
For train_loader, I think here should be `elf.cfg.batch_size_per_gpu` instead of `self.cfg.batch_size_val_per_gpu`?

File: pointcept/engines/hooks/evaluator.py (hunk 3 above)
Comment (Gofinge) [reply pointcept/engines/hooks/evaluator.py:190]: I think these are some unnecessary details which might confuse our user, especially for tensorboard/wandb, especially when we have hundreds of categories

File: configs/scannetpp/semseg-spunet-v1m1-0-base.py (hunk 1 above)
Comment (Gofinge) [reply configs/scannetpp/semseg-spunet-v1m1-0-base.py:25]: Yeah, I will put this change in another file, as this is a new setting different from the original paper. For base config, let's respect the setting from original paper. (I will make the modification)

**Review by Gofinge — APPROVED** [review]
